Sources/Controllers/BasicAudioController.swift

```diff
@@ -143,30 +187,23 @@ internal class AudioController: NSObject {
         }
     }
 
+    private func createAudioPlayer(url: URL) -> AVAudioPlayer? {
+        var audioPlayer: AVAudioPlayer?
+        do {
+            audioPlayer = try AVAudioPlayer(contentsOf: url)
+        } catch let error as NSError {
+            print("Failed to create AVAudioPLayer for url: \(url),\n with error: \(error.localizedDescription)")
+        }
+        return audioPlayer
+    }
+
 }
 
 // MARK: - AVAudioPlayerDelegate
-extension AudioController: AVAudioPlayerDelegate {
-
-    internal func audioPlayerDidFinishPlaying(_ player: AVAudioPlayer, successfully flag: Bool) {
-        // get audio item from message
-        var audioItem: AudioItem? = nil
-        switch playingMessage?.kind {
-        case .audio(let item)?: audioItem = item
-        default: break
-        }
-        // update cell UI
-        playingCell?.progressView.progress = 0.0
-        playingCell?.playButton.isSelected = false
-        if let item = audioItem {
-            playingCell?.durationLabel.text = AudioMessageCell.durationString(from: item.duration)
-        }
-        // reset properties since audio finished
-        progressTimer?.invalidate()
-        progressTimer = nil
-        audioPlayer = nil
-        playingMessage = nil
-        playingCell = nil
+extension BasicAudioController: AVAudioPlayerDelegate {
+
+    open func audioPlayerDidFinishPlaying(_ player: AVAudioPlayer, successfully flag: Bool) {
+        self.stopAnyOngoingPLaying()
     }
```

Comment:
I would add the other delegate methods for the case where there is some audio playing corruption